grails-app/domain/com/unifina/domain/data/Stream.groovy

```diff
@@ -155,4 +155,31 @@ class Stream implements Comparable {
 			return latestDataTimestamp.before(threshold)
 		}
 	}
+
+	/**
+	 * Return true if this Stream is an UI Channel.
+	 */
+	@CompileStatic
+	boolean isUIChannel() {
+		return uiChannel && uiChannelCanvas != null && uiChannelPath != null
+	}
+
+	/**
+	 * Return module id parsed from uiChannelPath or 0 if not found.
+	 */
+	@CompileStatic
+	Integer parseModuleID() {
+		if (uiChannelPath == null) {
+			return 0
```

Comment:
`0` is a valid result, so the caller can't know whether the call succeeded or failed. Better to throw `NullPointerException` or `InvalidStateException` (or similar)?